x/meshsecurity/client/cli/gov_tx.go

```
@@ -0,0 +1,136 @@
+)
+
+// DefaultGovAuthority is set to the gov module address.
+// Extension point for chains to overwrite the default
+var DefaultGovAuthority = sdk.AccAddress(address.Module("gov"))
+
+func SubmitProposalCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "submit-proposal",
+		Short:        "Submit a mesh security proposal.",
+		SilenceUsage: true,
+	}
+	cmd.AddCommand(
+		ProposalSetVirtualStakingMaxCapCmd(),
+	)
+	return cmd
+}
+
+func ProposalSetVirtualStakingMaxCapCmd() *cobra.Command {
+	bech32Prefix := sdk.GetConfig().GetBech32AccountAddrPrefix()
+	cmd := &cobra.Command{
+		Use:   "set-virtual-staking-max-cap [contract_addr_bech32] [max_cap] --title [text] --summary [text] --authority [address]",
+		Short: "Submit a set virtual staking max cap proposal",
+		Args:  cobra.ExactArgs(2),
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Submit a proposal to set virtual staking maximum cap limit to the given contract.
+
+Example:
+$ %s tx meshsecurity submit-proposal set-virtual-staking-max-cap %s1l94ptufswr6v7qntax4m7nvn3jgf6k4gn2rknq 100stake --title "a title" --summary "a summary" --authority %s1eppn90g4eg7guvyhz0ayph40c3e6uktygdd2nq
+`, version.AppName, bech32Prefix, bech32Prefix)),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, proposalTitle, summary, deposit, err := getProposalInfo(cmd)
+			if err != nil {
+				return err
+			}
+			authority, err := cmd.Flags().GetString(flagAuthority)
+			if err != nil {
+				return fmt.Errorf("authority: %s", err)
+			}
+
+			if len(authority) == 0 {
+				return errors.New("authority address is required")
+			}
+
+			src, err := parseSetVirtualStakingMaxCapArgs(args, authority)
+			if err != nil {
+				return err
+			}
+
+			proposalMsg, err := v1.NewMsgSubmitProposal([]sdk.Msg{&src}, deposit, clientCtx.GetFromAddress().String(), "", proposalTitle, summary)
+			if err != nil {
+				return err
+			}
+			if err = proposalMsg.ValidateBasic(); err != nil {
```

nice check to fail fast